web/src/layers/EditableIntersectionLayer.svelte

```diff
@@ -0,0 +1,97 @@
+<script lang="ts">
+  import { getContext } from "svelte";
+  import {
+    CircleLayer,
+    hoverStateFilter,
+    Popup,
+    SymbolLayer,
+  } from "svelte-maplibre";
+  import { layerId, mapMetersToPixels } from "../common";
+  import { Style } from "../common/colors";
+  import {
+    Intersection,
+    type IntersectionFeature,
+  } from "../common/Intersection";
+  import EditIntersectionPopup from "../edit/EditIntersectionPopup.svelte";
+  import type { RenderNeighbourhoodOutput } from "../wasm";
+
+  export let neighbourhood: RenderNeighbourhoodOutput =
+    getContext("neighbourhoodGj");
+  export let onClickIntersection = (intersection: Intersection) => {};
+  export let interactive: boolean = false;
+
+  /// NOTE: this takes the intersection's index in the neighborhood FeatureCollection, *not* the intersectionId!
+  function getIntersectionByFeatureIndex(
+    intersectionIdx: number,
+  ): Intersection {
+    // The Feature from maplibre has its nested properties serialized as a JSON string.
+    // see: https://github.com/dimfeld/svelte-maplibre/discussions/162#discussioncomment-9295264
+    //
+    // So we refetch the corresponding IntersectionFeature from the neighbourhood to get all it's properties.
+    let intersectionFeature = neighbourhood.features[intersectionIdx];
+    let intersection = new Intersection(
+      intersectionFeature as IntersectionFeature,
+    );
+    return intersection;
+  }
+
+  function getIntersectionFromFeatures(features: any): Intersection {
+    // Note: this `id` is autoassigned by maplibre, and corresponds to the index of the feature in the neighbourhood.
+    return getIntersectionByFeatureIndex(features[0].id);
+  }
+</script>
+
+<!-- 
```

Comment:
Just checked, this is a pre-existing problem clicking on a road too. If you click and really click again to delete the new filter, the map zooms in.

`map.doubleClickZoom.disable()` should fix this. But... we'd have to blanket disable this while in edit mode entirely. Not sure what the right approach is. It'd be nice to conditionally block this behavior, depending if a layer is double clicked, or the map is.

Reply:
Yeah, I was hoping that there was something like `event.preventDefault()`, but as of yet I haven't found it.

Reply:
I think disabling double click entirely while in edit mode would be a cure worse than the disease. At least it's clear whats happening now. The user can zoom back out. 

It is annoying though. Are you OK to merge without addressing this?

Reply:
Fine to ignore this problem for now